panic/src/panicking.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+#[allow(dead_code)]
+
+/// Common backend panic implementation
+///
+/// This is a subset of the functionality available in Rust's std
+/// [panicking.rs](https://github.com/rust-lang/rust/blob/master/library/std/src/panicking.rs)
+/// module.
+
+pub(crate) mod panic_count {
+    //! Number of panics that are currently being handled on the current thread
+    //!
+    //! This deviates from
+    //! [panicking.rs](https://github.com/rust-lang/rust/blob/master/library/std/src/panicking.rs)
+    //! It does not:
+    //! - use `GLOBAL_PANIC_COUNT` count. The global count is meant to provide
+    //!   an optimization on some platforms. It's not clear if SGX benefits from
+    //!   this optimization and thus needing this complexity
+    //! - no `set_always_abort()`. The docs say this is for `libc::fork`.
+    //!   Enclaves do not support forking or spawning child threads or processes
+    //! - use the `thread_local!` macro or the `::std::thread::LocalKey` type.
+    //!   These are generic over any type that `needs_drop`, but
+    //!   `usize` does not need drop
+    use core::cell::Cell;
+
+    /// Panic count for the current thread.
+    #[thread_local]
+    static LOCAL_PANIC_COUNT: Cell<usize> = Cell::new(0);
```

Comment:
Why not [AtomicUsize](https://doc.rust-lang.org/core/sync/atomic/struct.AtomicUsize.html)